functions/logStripeEvent.js

```diff
@@ -0,0 +1,90 @@
+const functions = require('firebase-functions');
+const admin = require('firebase-admin');
+const logDonation = require('./logDonation');
+if (admin.apps.length === 0) {
+  admin.initializeApp();
+}
+
+const DEFAULTCPL = 1.0;
+
+exports.logPaymentIntent = functions.https.onRequest(async (req, res) => {
+  if (!req.body) {
+    res.status(501).send('no data supplied!');
+    return;
+  }
+  let event;
+  try {
+    event = req.body;
+  } catch (err) {
+    console.log(`encountered error while parsing request: ${err}`);
+    res.status(501).send({msg: 'bad data supplied', data: req.body});
+    return;
+  }
+  console.log(`parsing event with id ${event.id}`);
+  switch (event.type) {
+    case 'payment_intent.succeeded':
+      const paymentIntent = event.data.object;
+      if (paymentIntent.description === 'Give Lively / Smart Donations') {
+        console.log(`successful payment for ${paymentIntent.amount}`);
+        this.handlePaymentIntentSucceeded(paymentIntent, event.id);
+      }
+      break;
+    default:
+      // ignore other events
+      break;
+  }
+  res.status(200).send({msg: 'sucessfully received event', data: event});
```

Comment:
You can return the `res.status` function directly `return res.status(200)...` without needing to specify `return` on the following line.